pkg/auth/providers/saml/saml_client.go

```diff
@@ -222,15 +229,23 @@ func (s *Provider) getSamlPrincipals(config *v3.SamlConfig, samlData map[string]
 
 	groups, ok := samlData[config.GroupsField]
 	if ok {
-		for _, group := range groups {
-			group := v3.Principal{
-				ObjectMeta:    metav1.ObjectMeta{Name: s.groupType + "://" + group},
-				DisplayName:   group,
-				Provider:      s.name,
-				PrincipalType: "group",
-				MemberOf:      true,
+		if s.hasLdapGroupSearch() {
+			var err error
+			groupPrincipals, err = s.ldapProvider.SearchPrincipals(userPrincipal.ObjectMeta.Name, "group", v3.Token{})
```

Comment:
This will eventually call https://github.com/rancher/rancher/blob/master/pkg/auth/providers/ldap/ldap_client.go#L359
Which means we will search for a group by the name `userPrincipal.ObjectMeta.Name`
Instead we should call SearchPrincipals api on each of the groups that we get from the saml assertion